File: plugins/dashboard/frontend/src/app/utils/confirmation_state.ts
export enum ConfirmationState {
    Undefined = 0,
    Rejected,
    Pending,
    Accepted,
    Confirmed,
}

export function resolveConfirmationState(sigType: number) {
    switch (sigType) {
        case ConfirmationState.Undefined:
            return "Undefined";
        case ConfirmationState.Rejected:
            return "Rejected";
        case ConfirmationState.Pending:
            return "Pending";
        case ConfirmationState.Accepted:
            return "Accepted";
        case ConfirmationState.Confirmed:
            return "Confirmed";
        default:
            return "Undefined Confirmation State";
    }
}

File: packages/protocol/engine/ledger/mempool/newconflictdag/confirmationstate.go
package newconflictdag

import (
	"strconv"
)

// ConfirmationState is the confirmation state of an entity.
type ConfirmationState uint8

func (c ConfirmationState) String() string {
	switch c {
	case Pending:
		return "Pending"
	case Accepted:
		return "Accepted"
	case Rejected:
		return "Rejected"
	default:
		return "Unknown (" + strconv.Itoa(int(c)) + ")"
	}
}

const (
	// Pending is the default confirmation state.
	Pending ConfirmationState = iota

	// Accepted is the state for accepted entities.
	Accepted

	// Rejected is the state for confirmed entities.
	Rejected
)
